### PR TITLE
rust-analyzer: 2021-04-05 -> 2021-04-12

### PR DESCRIPTION
**Pull request by Riey**
###### Motivation for this change

Bump version

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/rust-analyzer/default.nix [hunk 1]
```diff
@@ -5,19 +5,19 @@
 }:
 
 let
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/rust-analyzer/default.nix:7]:
```suggestion
```

File: pkgs/development/tools/rust/rust-analyzer/default.nix [hunk 2]
```diff
@@ -5,19 +5,19 @@
...
-  rev = "2021-04-05";
+  rev = "2021-04-12";
 in
 
 rustPlatform.buildRustPackage {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/rust-analyzer/default.nix:11]:
```suggestion
rustPlatform.buildRustPackage rec {
```

File: pkgs/development/tools/rust/rust-analyzer/default.nix [hunk 3]
```diff
@@ -5,19 +5,19 @@
...
   pname = "rust-analyzer-unwrapped";
   version = "unstable-${rev}";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/rust-analyzer/default.nix:13]:
```suggestion
  version = "2021-04-12";
```

File: pkgs/development/tools/rust/rust-analyzer/default.nix [hunk 4]
```diff
@@ -5,19 +5,19 @@
...
-  cargoSha256 = "sha256-kDwdKa08E0h24lOOa7ALeNqHlMjMry/ru1qwCIyKmuE=";
+  cargoSha256 = "1kzpymapkyzlmwy1lwyk3lnpg7y565j1m11z4xc8m4qyh1rdh8vb";
 
   src = fetchFromGitHub {
     owner = "rust-analyzer";
     repo = "rust-analyzer";
     inherit rev;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/rust-analyzer/default.nix:19]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
unstable in the version is wrong here as this is actually the release version.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119211` at d05c84d8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
  </ul>
</details>

**Review by oxalica — CHANGES_REQUESTED** [review]
Maybe also fix update script?
https://github.com/NixOS/nixpkgs/blob/f4e83361c2618fcdbbb5b07d11535fb8660de280/pkgs/development/tools/rust/rust-analyzer/update.sh#L15

https://github.com/NixOS/nixpkgs/blob/f4e83361c2618fcdbbb5b07d11535fb8660de280/pkgs/development/tools/rust/rust-analyzer/update.sh#L27

**Comment by Riey** [comment]
@oxalica should I also squash that patch into version up?

**Comment by oxalica** [comment]
@Riey 
> should I also squash that patch into version up commit?

I prefer to squash.

**Review by oxalica — APPROVED** [review]
The extension is built and works fine. Update script also works.

**Pull request merged**